### PR TITLE
fix for  FITS I/O bug #4580 

### PR DESCRIPTION
**Pull request by AMAN3003**
warning can be removed but screen looks tidy due to warning

### CODE REVIEW

**Comment by pllim** [comment]
Generally, each bug fix should be a separate PR. It is easier to review and backtrack that way.

**Comment by AMAN3003** [comment]
@plim you are right  next time i will take it in consider. could you guide how to remove this travis-ci error

**Comment by pllim** [comment]
```
astropy/io/ascii/html.py:22:3: E111 indentation is not a multiple of four
astropy/io/ascii/html.py:24:3: E111 indentation is not a multiple of four
astropy/io/ascii/html.py:26:3: E111 indentation is not a multiple of four
astropy/io/ascii/html.py:94:15: E111 indentation is not a multiple of four
astropy/io/ascii/html.py:96:15: E111 indentation is not a multiple of four
```

Indentation needs to be four spaces (not tab).

**Comment by MSeifert04** [comment]
see https://github.com/astropy/astropy/pull/4569#issuecomment-180124878

**Comment by AMAN3003** [comment]
@plim i was asking for the travis ci error of https://github.com/AMAN3003/astropy/commit/bf3baf45e1b601ed748e60638166b5f0a43145ff

**Comment by AMAN3003** [comment]
@MSeifert04 i was asking for the travis ci error for https://github.com/AMAN3003/astropy/commit/bf3baf45e1b601ed748e60638166b5f0a43145ff

could you tell how to fix the build problems ..or any refrence to different types of travis ci-error and its fix

File: fix_4580.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from astropy.io.fits.hdu import PrimaryHDU,BinTableHDU,HDUList
+dum = np.empty((5,100), dtype=[('w','>f8'),('f','>f4')])
+prihdu = fits.PrimaryHDU()
+tblhdu = fits.BinTableHDU(dum, name='DATA')
+hdulist = fits.HDUList([prihdu, tblhdu])
+hdulist.writeto('file.fits', clobber=True) # Does not fail
+warnings.filterwarnings('ignore', category=UserWarning, append=True)
+tmp = fits.open('jhfaj.fits',ignore_missing_end=True)
+
+
```
Comment (hamogu) [inline fix_4580.py]: It seems that this file has nothing to do with the rest of astropy. It's not a test, it's not something that a user would import.
Maybe I don't understand, but why it is here?

File: sample.tex [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
\put(80,70){\circle{30}}  \put(80,70){\circle*{10}}   % right eye
\put(40,40){\line(1,2){10}} \put(60,40){\line(-1,2){10}} \put(40,40){\line(1,0){20}} % nose
\put(50,20){\oval(80,10)[b]} % mouth
\multiput(0,90)(4,0){10}{\line(1,3){4}}  % left eyebrow
\multiput(100,90)(-4,0){10}{\line(-1,3){4}}  % right eyebrow
\end{picture}
\end{center}


\end{document}
```
Comment (hamogu) [inline sample.tex]: Any particular reason why this file was changed?

File: astropy/units/format/latex.py [hunk 3]
```diff
@@ -101,8 +101,8 @@ def format_exponential_notation(cls, val):
         -------
         latex_string : str
             The value in exponential notation in a format suitable for LaTeX.
-        """
-        if np.isfinite(val):
+        """#np.any() and np.all() which to use depends on the val array
+        if np.any(np.isfinite(val)):   #np.isfinite returns a booleans of array type so np.any(np.ndarray) returns true if any vaues in np.ndarray is true
```
Comment (hamogu) [inline astropy/units/format/latex.py]: I don't think the comments are needed here - np.any etc. are already explained in the numpy documentation, we don't have to repeat that here.

File: astropy/io/ascii/html.py [hunk 4]
```diff
@@ -83,7 +90,10 @@ def process_lines(self, lines):
                                         'installed to read HTML tables')
 
         if 'parser' not in self.html:
-            soup = BeautifulSoup('\n'.join(lines))
+            if lxml_installed:
+                soup = BeautifulSoup('\n'.join(lines),"lxml")
+            else:
+                soup = BeautifulSoup('\n'.join(lines),"html.parser")
```
Comment (hamogu) [inline astropy/io/ascii/html.py]: This is a good way of doing it! Just a tiny comment. There should be a space after the comma in the arguments.

**Comment by embray** [comment]
This PR is combining a whole mess of different issues, and does not even begin to fix the actual bug it's proposing to fix, sorry.

**Pull request closed without merging**